src/coreclr/nativeaot/Runtime/unix/PalRedhawkUnix.cpp

```diff
@@ -481,10 +483,10 @@ EXTERN_C intptr_t RhGetCurrentThunkContext()
 //  thread        - thread to attach
 extern "C" void PalAttachThread(void* thread)
 {
-#if HAVE_THREAD_LOCAL
-    tls_destructionMonitor.SetThread(thread);
+#ifdef TARGET_LINUX
+    pthread_setspecific(key, thread);
```

Comment:
Error handling? We probably just want to fail-fast when this fails. There is no good way to recover.